Add email functionality

TeddySmoker opened the pull request:
Hello, 

As I've seen that this example does not have an have any email functionality yet I've decided to implement it myself from what I've learned in the past week by using openmaize, I love it by the way, and bamboo. 

This feature is implemented by following the Phoenix Framework docs (http://www.phoenixframework.org/docs/sending-email).

Beside adding the email functionality I've also changed a bit the controllers that were using the Mailer functions in order to make the tests pass. Haven't wrote any new tests but I can if it might be necessary. 

This should be fairly easy for other people to implement themselves but I can add some more documentation if needed.

riverrun commented:
Looks cool. Thanks.
I just want some time to look at this change in more depth and then I'll get back to you.

TeddySmoker commented:
Alrighty.

I was in doubts of splitting Mailer into an extra module (email) but I went with it as the docs 'recommended' it.

I am considering of refactoring it so the email functions reside in Mailer, just as it was before, so it is less confusing to other people.

riverrun commented:
Let me look at it first. I should have time today or tomorrow.

riverrun commented on config/config.exs:
```diff
@@ -17,6 +17,20 @@ config :welcome, Welcome.Endpoint,
+  adapter: Bamboo.SMTPAdapter,
+  server: "smtp.domain",
+  port: 1025,
+  username: "username",
```
Let's use the `SYSTEM.get_env` values for username and password

riverrun commented on lib/email.ex:
```diff
@@ -0,0 +1,45 @@
+    |> to(email)
+    |> from("welcome@example.com")
+    |> subject("Confirm your account - Welcome Example")
+    |> text_body("Confirm your Welcome Example email here http://www.example.com/sessions/confirm?#{link}")
```
Do we need to add ` |> Mailer.deliver_now` to these three functions?

TeddySmoker replied:
No,
We pipe the function in the controller like this  `Email.ask_confirm(email, link) |> Mailer.deliver_now`
I don't know how I forgot to add that. I will fix the typos and do the needed changes then commit again.

TeddySmoker replied:
On a second thought, you might be right about adding `|> Mailer.deliver_now` to those three functions, do you recon it will work the same way as adding the `|> Mailer.deliver_now` to the controller? 

It makes sense now thinking that `plug Openmaize.ConfirmEmail` uses an Email function to which I can't add `|> Mailer.deliver_now`

riverrun commented on web/controllers/password_reset_controller.ex:
```diff
@@ -17,7 +17,7 @@ defmodule Welcome.PasswordResetController do
     case Repo.update(changeset) do
       {:ok, _user} ->
-        Mailer.ask_reset(email, link)
+        Mail.ask_reset(email, link)
```
I think this should be `Email`, not `Mail`

TeddySmoker commented:
Hey! 
Sorry for taking so long, these past days have been a nightmare. 

So here is what I've done. 
I've fixed the typos / bugs you've mentioned me and replaced the username/password with System.get_env. 

I've also added `Mailer.deliver_now` to the functions that were using the Email functions, the only place I couldn't add it to was `plug Openmaize.ConfirmEmai'l`. 
I don't know how I should proceed on that one, any tips?

riverrun commented:
Merged into new_mailer branch as there are still a couple of things to work on.

Thanks for your help. I'll be merging this into master soon.
